sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/DoFnSignatures.java

```diff
@@ -299,21 +299,49 @@ private static Method findAnnotatedMethod(
     }
 
     // We need to be able to call it. We require it is public.
-    checkArgument(
+    errors.checkArgument(
         (first.getModifiers() & Modifier.PUBLIC) != 0, "%s must be public", format(first));
 
     // And make sure its not static.
-    checkArgument(
+    errors.checkArgument(
         (first.getModifiers() & Modifier.STATIC) == 0, "%s must not be static", format(first));
 
     return first;
   }
 
   private static String format(Method m) {
-    return ReflectHelpers.CLASS_AND_METHOD_FORMATTER.apply(m);
+    return (m == null) ? "null" : ReflectHelpers.CLASS_AND_METHOD_FORMATTER.apply(m);
```

Comment:
This format logic could move to the `ErrorReporter`, right? Shouldn't that be the only place that needs to format methods?